### PR TITLE
Clean up COM interfaces

### PR DESCRIPTION
**Pull request by NextTurn**

### CODE REVIEW

**Comment by leculver** [comment]
I'd like to review this in depth so I may not get to it until I'm fully back.

File: src/Microsoft.Diagnostics.Runtime/src/Utilities/PEImage/ImageResourceDirectoryEntry.cs [hunk 1]
```diff
@@ -21,44 +22,42 @@ namespace Microsoft.Diagnostics.Runtime.Utilities
     /// set to indicate this.  Otherwise the high bit is clear and the offset
     /// field points to a resource data entry.
     /// </summary>
-    internal struct IMAGE_RESOURCE_DIRECTORY_ENTRY
+    [StructLayout(LayoutKind.Sequential)]
```
Comment (leculver) [inline src/Microsoft.Diagnostics.Runtime/src/Utilities/PEImage/ImageResourceDirectoryEntry.cs]: Has this struct change been tested?  Is the layout before/after the `StructLayout` and field rearrange the same?

Comment (leculver) [reply src/Microsoft.Diagnostics.Runtime/src/Utilities/PEImage/ImageResourceDirectoryEntry.cs]: Ah these are the only fields, the rest are properties.

Comment (NextTurn) [reply src/Microsoft.Diagnostics.Runtime/src/Utilities/PEImage/ImageResourceDirectoryEntry.cs]: Struct layouts are sequential by default.

See Remarks in https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.structlayoutattribute

**Pull request merged**